Rename addr to address in non-public API.

jiminhsieh opened the pull request:
Issue - #21874
I will use another commit to change attributes from `addr` to `address` in Java. I found previous PR, #21885, not only simply change  `addr` to `address` but also **methods** and attributes contained `addr` in the middle. For instance, `getAddrString` to `getAddressString`. Is it fine for the Akka-Team? Thanks!

johanandren reviewed (approved):
Not a huge improvement, but LGTM, I guess.

johanandren commented:
OK TO TEST

johanandren commented:
Failure is formatting, the code is auto formatted when compiled, so please compile all changed code locally and then commit the changes and push that.

jiminhsieh commented:
@johanandren Really sorry for that stupid mistake. I only used IntelliJ in local without sbt. By the way, can I verify broke public API in the local with `sbt test`? Thanks!

raboof commented:
@jiminhsieh for checking binary compatibility have a look at https://github.com/akka/akka/blob/master/CONTRIBUTING.md#binary-compatibility

johanandren commented:
Seems this caused a few test failures now, plus binary compatibility errors.

See https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/9885/ for details.

jiminhsieh committed:
Rename addr to full phrase.

johanandren commented on akka-actor/src/main/scala/akka/actor/ActorPath.scala:
```diff
@@ -272,11 +272,11 @@ final case class RootActorPath(address: Address, name: String = "/") extends Act
   override val toSerializationFormat: String = toString
 
-  override def toStringWithAddress(addr: Address): String =
+  override def toStringWithAddress(newAddress: Address): String =
```
These are actually not source compatible, since the parameter names in Scala are a part of the API (because you could use named params, not likely that anyone does for this but we cannot know nobody did a `toStringWithAddress(addr = something)` which will not work anymore with this change.

Strange that MiMa didn't complain about that.

jiminhsieh replied:
Ok, I got it. Thanks for you explanation! : ) I will rollback this soon. BTW, maybe it would be a good idea to send an issue to MiMa? They would know this situation.

johanandren replied:
Ping @ktoso or @2m is this perhaps source incompatible but MiMa only cares about binary incompatibilities?

ktoso replied:
Yeah this is breaking only for scala in that specific use case. 
Scala has tools to deal with that though - the `def method(@deprecatedName("addr") newOne: Address)` syntax can make that work, though I'm unsure we care enough about this name here to make the change - let's not change it in method params perhaps?

ktoso replied:
Link to docs https://www.scala-lang.org/api/current/scala/deprecatedName.html

jiminhsieh replied:
Does test code need to be consider compatibility? Normally, it would be to fine to change public methods or public attributes in test code. Thanks!

ktoso replied:
Tests as well as the testkit (which is public API but exempt from these rules) are fine to change :)

jiminhsieh commented:
I will review the compatibility of each file again. Since there are some use cases MiMa could not detect.

jiminhsieh committed:
Rollback to original naming.
It will break source compatibility. For example, toStringWithAddress(addr = something).

johanandren reviewed (approved):
LGTM

patriknw reviewed (approved):
LGTM
